### PR TITLE
DNSName logging fixes

### LINKED ISSUE #3589: After #3569 may want to review DNSName logging
#3569 introduces `DNSName::toLogString`, we probably want to use this in lots of places that right now either choke on `empty` or check explicitly and verbosely.

### LINKED ISSUE #4002: replace all toString() and <<DNSName<< in logging to toLogString()
to avoid throwing on empty DNSNames

### PR DESCRIPTION
**Pull request by pieterlexis**
This PR removes the inconsistent use of `DNSName.toString()`, `DNSName.toStringNoDot()`, `DNSName.toStringRootDot()` and `DNSName.toLogString()` with regards to logging.

It makes the `<<` operator for DNSNames call `toLogString()`, which in turn uses `toStringRootDot()` on a non-empty DNSName. The output provided should therefor be consistent with the pre-DNSName logging, and print root always as '.'.

This PR also fixes dot-correctness to the `zone2json` tool and "root zone is a dot" semantics to the pipebackend (as is already the case with the gsql backends).

Closes #4002 
Closes #3589

### CODE REVIEW

File: pdns/dnsbackend.cc [hunk 1]
```diff
@@ -357,7 +357,7 @@ string serializeSOAData(const SOAData &d)
 {
   ostringstream o;
   //  nameservername hostmaster serial-number [refresh [retry [expire [ minimum] ] ] ]
-  o<<d.nameserver.toString()<<" "<< d.hostmaster.toString() <<" "<< d.serial <<" "<< d.refresh << " "<< d.retry << " "<< d.expire << " "<< d.default_ttl;
+  o<<d.nameserver<<" "<< d.hostmaster <<" "<< d.serial <<" "<< d.refresh << " "<< d.retry << " "<< d.expire << " "<< d.default_ttl;
```
Comment (Habbie) [inline pdns/dnsbackend.cc]: given that ostreams now use `toLogString`, this code suddenly no longer throws while I think it should

File: docs/markdown/authoritative/backend-pipe.md [hunk 2]
```diff
@@ -69,6 +69,7 @@ If this time is ever exceeded, the backend is declared dead and a new process is
 If set, only questions matching this regular expression are even sent to the backend.
 This makes sure that most of PowerDNS does not slow down if you you deploy a slow backend.
 A query for 'www.powerdns.com' would be presented to the regex as 'www.powerdns.com', a matching regex would be '^www.powerdns.com$'.
+**Note**: to match the root domain, use a dot, e.g. '^.$'
```
Comment (Habbie) [inline docs/markdown/authoritative/backend-pipe.md]: I think this dot needs escaping

File: pdns/dnsbackend.cc (hunk 1 above)
Comment (pieterlexis) [inline pdns/dnsbackend.cc]: This function is never wrapped directly inside a try/catch. But now that you mention it.. broken SOA data should indeed be not allowed.

File: docs/markdown/authoritative/backend-pipe.md (hunk 2 above)
Comment (pieterlexis) [inline docs/markdown/authoritative/backend-pipe.md]: yes, also the other dots above it (we do indeed use Regex here)

**Comment by Habbie** [comment]
Review done. Few nits.

**Comment by pieterlexis** [comment]
nits fixed

**Pull request merged**